### PR TITLE
fix(textfield): add rows attribute

### LINKED ISSUE #3331: [Feat]: Add rows attribute to enable multiline Textarea with a predefined amount of visible lines
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Textfield

### Description of the requested feature

We are in the process to replace all `react-spectrum` with `spectrum-web-component` for the Share Dialog component. The Share Dialog uses a multiline Textarea with a predefined amount of rows. To replicate the current experience of the Share Dialog, we need a way to set the number of visible rows for the Textarea.

### Mockups or screenshots

image

### Implementation notes or ideas

The change would require to pass along the `rows` attribute to the underlying `<textarea>` and adjust the CSS to allow the textarea to grow to show the rows and to disable the resizing capability

### PR DESCRIPTION
**Pull request by ingorichter**
## Description

Adding the rows attribute to enable a multiline Textarea to show up with a predefined amount of rows. We need this feature for an existing feature where we replace all react-spectrum components with spectrum-web-components.

## Related issue(s)

- fix #3331

## Motivation and context

Product required for Adobe Share Dialog

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Screenshots (if appropriate)

image

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/textfield/src/textfield.css [hunk 1]
```diff
@@ -153,6 +153,17 @@ textarea {
...
 }
 
+:host([multiline][rows]) .input {
+    block-size: auto;
+    resize: none;
+}
+
+:host([multiline][rows="1"]) .input {
+    block-size: auto;
+    resize: none;
```
Comment (Westbrook) [inline packages/textfield/src/textfield.css:163]: These lines should already be covered by line 157/8 from above. Can we remove?

File: packages/textfield/src/Textfield.ts [hunk 2]
```diff
@@ -97,6 +97,9 @@ export class TextfieldBase extends ManageHelpText(Focusable) {
     @property({ type: Boolean, reflect: true })
     public readonly = false;
 
+    @property({ type: Number })
+    public rows = -1;
```
Comment (Westbrook) [inline packages/textfield/src/Textfield.ts:101]: This makes sense to normalize across the file, thanks for calling that out!

**Review by Westbrook — COMMENTED** [review]
This looks good to me. We've got one piece left and we'll be ready to merge. Can you checkout the [docs on Visual Regression Testing](https://github.com/adobe/spectrum-web-components#keeping-ci-assets-updated) and update the golden images hash? Feel free to message me here or in Slack if any of that is unclear.

Thanks!

File: packages/textfield/src/textfield.css (hunk 1 above)
Comment (ingorichter) [reply packages/textfield/src/textfield.css]: Yes, I cleaned it up. Thanks for spotting it

**Review by Westbrook — APPROVED** [review]
Thanks @ingorichter this looks good to me. I can't guarantee a pre-shutdown release, but if not, look for this to be available shortly thereafter. Cheers!

**Pull request merged**